### PR TITLE
WIP: feat: crud workspaces

### PR DESCRIPTION
**Pull request by rodrimaia**
#64 

https://www.loom.com/share/269af08124544e81a5e58471ffe2a67c 

- [x] View Workspaces:
  - we are displaying two views: My Workspaces and All Workspaces (the user can still use the quickPick command to filter and open all workspaces)
  - We have also a context menu showing a link to the workspace dashboard page 
- [x] Create Workspaces:
  - A button pointing to the dashboard/create extension page

image

### CODE REVIEW

**Comment by BrunoQuaresma** [comment]
I see some plugins asking to confirm the "name" of the thing you want to delete. Probably we want to do the same. What do you think?

**Review by bpmct — COMMENTED** [review]
Any plans to add support for an "edit' button which goes to the workspace settings?

File: src/api-helper.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+): [agents: WorkspaceAgent[], folderPath: string | undefined] {
+  // TODO: multiple agent support
+  const agents = workspace.latest_build.resources.reduce((acc, resource) => {
+    return acc.concat(resource.agents || [])
+  }, [] as WorkspaceAgent[])
+
+  let folderPath = undefined
+  if (agents.length === 1) {
+    folderPath = agents[0].expanded_directory
+  }
```
Comment (Emyrk) [inline src/api-helper.ts:14]: If there is more than 1 agent, does everything fail? Should we randomly choose one?

File: src/commands.ts [hunk 2]
```diff
@@ -79,6 +81,9 @@ export class Commands {
         throw new Error("Failed to get authenticated user")
       }
       await vscode.commands.executeCommand("setContext", "coder.authenticated", true)
+      if (user.roles.find((role) => role.name === "owner")) {
+        await vscode.commands.executeCommand("setContext", "coder.isOwner", true)
+      }
```
Comment (Emyrk) [inline src/commands.ts:86]: Why do we need to check for the owner role? We prefer to do capability checks. You can send a payload to `https://dev.coder.com/api/v2/authcheck` to see if you can do something. Eg:
```json
{
   "checks":{
      "readAllUsers":{
         "object":{
            "resource_type":"user"
         },
         "action":"read"
      },
   },
}
```

Is this just to see if we have arbitrary execution of all workspaces?

Comment (Emyrk) [reply src/commands.ts:86]: I think this is ok for now, we but we might want to change this soon, especially with https://github.com/coder/coder/pull/6875

File: src/api-helper.ts (hunk 1 above)
Comment (rodrimaia) [reply src/api-helper.ts:14]: That is a good question. I am not changing the actual behavior, just extracted it to a new function. the current version also does this `agents[0]` choice =/ . I am not sure what is the best way to handle multiple agents...

File: src/commands.ts (hunk 2 above)
Comment (rodrimaia) [reply src/commands.ts:86]: We decided to only show the "all Workspaces" view for users who have the owner role. sure we can revisit this after the owner connections PR

**Review by BrunoQuaresma — APPROVED** [review]
FE code LGTM

**Pull request merged**